csharp/ql/lib/ext/generated/System.Collections.model.yml

```diff
@@ -41,11 +41,16 @@ extensions:
       - ["System.Collections", "Hashtable", False, "Synchronized", "(System.Collections.Hashtable)", "", "Argument[0]", "ReturnValue.SyntheticField[System.Collections.Hashtable+SyncHashtable._table]", "value", "dfc-generated"]
       - ["System.Collections", "Hashtable", False, "get_EqualityComparer", "()", "", "Argument[this].SyntheticField[System.Collections.Hashtable._keycomparer]", "ReturnValue", "value", "dfc-generated"]
       - ["System.Collections", "ICollection", True, "get_SyncRoot", "()", "", "Argument[this]", "ReturnValue", "value", "dfc-generated"]
+      - ["System.Collections", "IComparer", True, "Compare", "(System.Object,System.Object)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["System.Collections", "IComparer", True, "Compare", "(System.Object,System.Object)", "", "Argument[1]", "Argument[this]", "taint", "df-generated"]
       - ["System.Collections", "IDictionary", True, "GetEnumerator", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
       - ["System.Collections", "IDictionaryEnumerator", True, "get_Entry", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
       - ["System.Collections", "IDictionaryEnumerator", True, "get_Key", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
       - ["System.Collections", "IDictionaryEnumerator", True, "get_Value", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
       - ["System.Collections", "IEnumerator", True, "get_Current", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
+      - ["System.Collections", "IEqualityComparer", True, "Equals", "(System.Object,System.Object)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
+      - ["System.Collections", "IEqualityComparer", True, "Equals", "(System.Object,System.Object)", "", "Argument[1]", "Argument[this]", "taint", "df-generated"]
+      - ["System.Collections", "IEqualityComparer", True, "GetHashCode", "(System.Object)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
```

Comment:
Do these summaries make sense? I think generally `Equals` would not propagate taint from arg to qualifier. Similarly I don't think `GetHashCode` should propagate flow either. There could be implementations that cache the arguments, and therefore taint is propagated to the qualifier, but that behavior doesn't match the general purpose of these methods.

Reply:
No, these are not very good summaries. The provenance of the models are `df-generated`, so these models come from the "old" heuristic for detecting models.
In this case, exactly as you guess, because there exists an implementation, where the heuristic derives that flow propgates to `this` (as there is a field containing a `Func` which is applied to the arguments) and then the model is lifted.
One could argue yet another improvement we could consider is *only* to lift models in case the models are *functional* - but I think that is out of scope for this PR.